zio-http/src/main/scala/zhttp/http/HttpData.scala

```diff
@@ -118,7 +117,7 @@ object HttpData {
 
   private[zhttp] final case class Text(text: String, charset: Charset) extends HttpData[Any, Nothing] with Cached
```

Comment:
Delete `Text` `Complete`

Reply:
We can change the approach a bit — Instead of adding any default headers via the `Response` constructors we could add them via the handler directly.

**Advantages**
1. You can be a bit more intelligent about what headers should be set.
2. You get the final response, ie. after all the middlewares etc. So you can make better assumptions about content-length and transfer-encoding etc.
3. For sending files (later) we can use the same technique, ie create a chunked writer instance and use that to write a file directly. 
4. We can do better memoization this way, specially for files.

Reply:
Done.
Also refactored Handler a bit, as there was code duplication.